x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/googleaistudio/GoogleAiStudioCompletionResponseEntity.java

```diff
@@ -77,33 +77,36 @@ public class GoogleAiStudioCompletionResponseEntity {
     public static ChatCompletionResults fromResponse(Request request, HttpResult response) throws IOException {
         var parserConfig = XContentParserConfiguration.EMPTY.withDeprecationHandler(LoggingDeprecationHandler.INSTANCE);
         try (XContentParser jsonParser = XContentFactory.xContent(XContentType.JSON).createParser(parserConfig, response.body())) {
-            moveToFirstToken(jsonParser);
+            return new ChatCompletionResults(List.of(new ChatCompletionResults.Result(content(jsonParser))));
+        }
+    }
 
-            XContentParser.Token token = jsonParser.currentToken();
-            ensureExpectedToken(XContentParser.Token.START_OBJECT, token, jsonParser);
+    public static String content(XContentParser jsonParser) throws IOException {
+        moveToFirstToken(jsonParser);
```

Comment:
Can the process in this function just be completed in a loop? Something like:
```
moveToFirstToken(jsonParser);
XContentParser.Token token = jsonParser.currentToken();

for(String field : List.of("candidates", "content", "parts", "text")) {
     ensureExpectedToken(XContentParser.Token.START_OBJECT, token, jsonParser);
     positionParserAtTokenAfterField(jsonParser, field, FAILED_TO_FIND_FIELD_TEMPLATE);
    jsonParser.nextToken();
}
XContentParser.Token contentToken = jsonParser.currentToken();
ensureExpectedToken(XContentParser.Token.VALUE_STRING, contentToken, jsonParser);
return jsonParser.text();
```

(Note: Can probably move the list of fields into a constant at the top of the class)

Reply:
That's a clever way to do it.  Can we follow up with that change, though? My intention was to not change the parsing logic as part of this commit, and instead reuse it as-is for streaming vs not-streaming